firestore/src/include/firebase/firestore/local_cache_settings.h

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIREBASE_FIRESTORE_SRC_INCLUDE_FIREBASE_FIRESTORE_LOCAL_CACHE_SETTINGS_H_
+#define FIREBASE_FIRESTORE_SRC_INCLUDE_FIREBASE_FIRESTORE_LOCAL_CACHE_SETTINGS_H_
+
+#include <cstdint>
+#include <memory>
+
+#include "Firestore/core/src/api/settings.h"
+#include "firestore/src/include/firebase/firestore/settings.h"
+#include "firestore/src/main/firestore_main.h"
+#include "firestore/src/main/local_cache_settings_main.h"
+
+namespace firebase {
+namespace firestore {
+
+class PersistentCacheSettingsInternal;
+class MemoryCacheSettingsInternal;
+
+/**
+ * Abstract class implemented by all supported cache settings.
+ *
+ * `PersistentCacheSettings` and `MemoryCacheSettings` are the only cache types
+ * supported by the SDK. Custom implementation is not supported.
+ */
+class LocalCacheSettings {
+ public:
+  virtual ~LocalCacheSettings() = default;
+
+  /** Equality function. */
+  friend bool operator==(const LocalCacheSettings& lhs,
+                         const LocalCacheSettings& rhs);
+
+ private:
+  friend class FirestoreInternal;
+  friend class Settings;
+
+  virtual api::LocalCacheSettings::Kind kind() const = 0;
+  virtual const api::LocalCacheSettings& core_cache_settings() const = 0;
+};
+
+/**
+ * Configures the SDK to use a persistent cache. Firestore documents and
+ * mutations are persisted across App restart.
+ *
+ * This is the default cache type unless explicitly specified otherwise.
+ *
+ * To use, create an instance using `PersistentCacheSettings::Create`, then
+ * pass it to an instance of `Settings` via `set_local_cache_settings()`, and
+ * use the `Settings` instance to configure the Firestore SDK.
+ */
+class PersistentCacheSettings final : public LocalCacheSettings {
+ public:
+  /** Create a default instance `PersistenceCacheSettings`. */
+  static PersistentCacheSettings Create();
```

Comment:
Should you make the default constructor private? I think the compiler-generated default constructor will be public, which doesn't seem to be desirable.